scripts/postprocess_sitemap.py

```diff
@@ -0,0 +1,25 @@
+from bs4 import BeautifulSoup
+
+with open('./_build/html/sitemap.xml', 'r') as f:
+    contents = f.read()
+
+soup = BeautifulSoup(contents, 'lxml')
+
+urls = soup.find_all('url')
+
+for url in urls:
+    loc = url.find('loc')
+    text = loc.string
+    # Remove the 'gen' and '404' pages
+    if '404' in text:
+        url.decompose()
+        continue
+    if 'genindex.html' in text:
```

Comment:
It would probably be better to be more specific that `genindex.html` needs to be at the end, since it then goes to strip off the last few characters (in *practice* I'm sure we'd never get `genindex.html` in the middle of a string, but it's probably better to be safer).
```suggestion
    if text.endswith('genindex.html'):
```

(but a good catch that this needs to be special-cased so we don't end up with `gen`)